MapboxCoreNavigation/RouteController.swift

```diff
@@ -34,15 +34,16 @@ open class RouteController: NSObject {
     }
     
     private var _routeProgress: RouteProgress {
+        willSet {
+            resetObservation(for: _routeProgress)
+        }
         didSet {
             movementsAwayFromRoute = 0
             updateNavigator(with: _routeProgress)
             updateObservation(for: _routeProgress)
```

Comment:
Shouldn't we not only subscribe to `legIndex` updates here, but also run the subscription handler too to update `RouteController with new `_routeProgress`?

Reply:
I'm not sure what you mean by this? `_routeProgress` is the memoized property that represent's `RouteController`'s copy of the current route progress.

Reply:
When `_routeProgress` is set, we observe it's `legIndex` to call `updateRouteLeg` when it is updated. But shouldn't we call `updateRouteLeg` right now, since we've just updated the entire `_routeProgress` and thus it's `legIndex` is technically updated too.

Reply:
@Udumft No, it's unnecessary, because the [leg is set](https://github.com/mapbox/mapbox-navigation-ios/pull/2352/files#diff-1a2149692b3d7c58b6a0392120d6011aR179) when we call `updateNavigator(with:)`.